### DISCUSSION TITLE
Controlling JSON schema of the autoform

### DISCUSSION THREAD

**Discussion by RajathShanmugaiah** [react-page/react-page#1091] [Q&A] [answered]
For eg:

```
 Renderer:({data}) =>{   
        return <Xcomponent xData={data}/>
    },
    id: 'xyz',
    title: 'X component',
    version: 1,
    controls: {
      type: 'autoform',      
      schema: {
        properties: {
            Id:{
                type:'string',
                default:'V1',
                enum: ['V1', 'V2', 'V3', 'V4', 'V5'],
            }, 
           image:{
                        type: 'string',
                        uniforms:{
                            component: ImageUrlField, fieldName: 'Section Image'
                        },
                        default: ''
                    },
     }
  }
}

```
What I want to do is have a **condition in image** which only **renders** when the **Id** is **V4** and is not shown in any other Id in the autoform. So is there a way to conditionally render the fields.

**Comment by zehawki** [comment, marked as answer]
Not supported by Uniforms via schema. That library is a bit of a joke. https://uniforms.tools/docs/examples-custom-fields/#displayif

**Reply by RajathShanmugaiah** [reply]
@zehawki  Thank you for your information. :D

**Comment by macrozone** [comment]
hi @RajathShanmugaiah, i added support for `showIf` in the schema,

see https://react-page.github.io/docs/#/custom-cell-plugins?id=passing-props-to-fields

example: https://github.com/react-page/react-page/blob/master/examples/pages/examples/conditionalForm.tsx#L38

**Reply by zehawki** [reply]
@macrozone you are da man.
